src/session_impl.cpp

```diff
@@ -640,6 +640,10 @@ namespace aux {
 			update_listen_interfaces();
 			open_listen_port();
 		}
+		
+		//call invariant check again after we did repeat the update calls
+		INVARIANT_CHECK;
```

Comment:
this will cause the invariant check to be run twice. it would be better to call the function directly (check_invariant()) but it would have to be wrapped in ``#ifdef`` to only be called when invariant checks are enabled in the build